src/util/map.js

```diff
@@ -20,13 +20,14 @@ export function toObject<K: string, V, InK: K, InV: V>(
  * pairs corresponding to the enumerable own properties of the object in
  * iteration order, as returned by `Object.keys`.
  */
-export function fromObject<K, V, InK: K & string, InV: V>(object: {
+export function fromObject<K, V, InK: K, InV: V>(object: {
```

Comment:
@wchargin This is the only change that probably needs to be rethought, since it breaks the static test below. When the intersection with string isn't removed, we now see this error:
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/util/map.js:30:22

Cannot get object[key] because intersection type [1] is incompatible with K [2]. [incompatible-use]

 [2] 23│ export function fromObject<K, V, InK: K & string, InV: V>(object: {
     24│   +[InK]: InV,
     25│ }): Map<K, V> {
     26│   const result = new Map();
 [1] 27│   const keys = (Object.keys(object): InK[]);
     28│   for (const key of keys) {
     29│     // FlowIgnore[incompatible-use]
     30│     const val: InV = object[key];
     31│     result.set(key, val);
     32│   }
     33│   return result;



Found 1 error
```

I think this is a bug, because I get different behavior (different errors) when I reverse the order of the types in the intersection.